lib/collection/src/lookup.rs

```diff
@@ -0,0 +1,76 @@
+use std::collections::HashMap;
+
+use futures::Future;
+use itertools::Itertools;
+use schemars::JsonSchema;
+use segment::data_types::groups::PseudoId;
+use segment::types::{PointIdType, WithPayloadInterface, WithVector};
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLockReadGuard;
+
+use crate::collection::Collection;
+use crate::operations::consistency_params::ReadConsistency;
+use crate::operations::types::{CollectionError, CollectionResult, PointRequest, Record};
+use crate::shards::shard::ShardId;
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(untagged)]
+pub enum Lookup {
+    None,
+    Single(Record),
+}
+
+impl From<Record> for Lookup {
+    fn from(record: Record) -> Self {
+        Lookup::Single(record)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct LookupRequest {
+    pub collection_name: String,
+    pub with_payload: WithPayloadInterface,
+    pub with_vectors: WithVector,
+}
+
+pub async fn lookup_ids<'a, F, Fut>(
+    request: LookupRequest,
+    values: Vec<PseudoId>,
+    collection_by_name: F,
+    read_consistency: Option<ReadConsistency>,
+    shard_selection: Option<ShardId>,
+) -> CollectionResult<HashMap<PseudoId, Lookup>>
+where
+    F: FnOnce(String) -> Fut,
+    Fut: Future<Output = Option<RwLockReadGuard<'a, Collection>>>,
+{
+    let collection = collection_by_name(request.collection_name.clone())
+        .await
+        .ok_or(CollectionError::NotFound {
+            what: format!("Collection {}", request.collection_name),
+        })?;
+
+    let ids = values
+        .into_iter()
+        .filter_map(|v| PointIdType::try_from(v).ok())
```

Comment:
Would that cool to somehow detect the case where `ids` is empty because the user input is incorrect.
e.g. I want to lookup but I provide the wrong payload key.